src/lib/containers/CardContainer.tsx

```diff
@@ -40,18 +40,53 @@ export type CardContainerProps = {
 } & CardConfiguration
 
 export const CardContainer = (props: CardContainerProps) => {
+
+  const [pageCount, setPageCount] = useState<number>(1)
+  const {
+    data,
+    limit = SynapseConstants.PAGE_SIZE,
+    isHeader = false,
+    facet,
+    unitDescription,
+    type,
+    isLoading,
+    secondaryLabelLimit = 3,
+    showBarChart = true,
+    title,
+    token,
+    getLastQueryRequest,
+    executeQueryRequest,
+    hasMoreData,
+    ...rest
+  } = props
+
+  let mounted = true
+
+  useEffect(() => {
+    if (mounted) {
+      const queryOffset = getLastQueryRequest!().query?.offset!
+      if (queryOffset === 0) {
+        setPageCount(1)
+      }
+    }
+    return () => {
+      mounted = false
+    }
+  }, [data])
+
   /**
    * Handle a click on next or previous
    *
    * @memberof SynapseTable
    */
   const handleViewMore = () => {
-    const queryRequest = props.getLastQueryRequest!()
+    const queryRequest = getLastQueryRequest!()
     let offset = queryRequest.query.offset!
     // paginate forward
-    offset += PAGE_SIZE
+    offset += SynapseConstants.PAGE_SIZE
```

Comment:
New offset ignores limit?  Expected it to use the configurable limit to figure out what the new offset of the query should be.